### DISCUSSION TITLE
Custom repository for the images used by the side services

### DISCUSSION THREAD

**Discussion by andreacoccodi** [jaegertracing/jaeger-operator#1489] [Q&A]
Hi, 
I'm looking to cache the jaegertracing images in our docker registry, and we found a way to do that for jaegertracing/jaeger-operator, but can't find anything that could apply to:
- jaegertracing/jaeger-es-index-cleaner:
- jaegertracing/jaeger-collector
- jaegertracing/jaeger-query
Any suggestion would be highly appreciated.
Thanks

**Comment by jpkrohling** [comment]
Could you clarify why it works for the jaeger-operator, but not for the other images? The caching mechanism should work for all of them, no?

**Reply by andreacoccodi** [reply]
values.yaml exposes the repository field, which applies only to the operator unfortunately
`jaeger-operator:
  nameOverride: jaeger-operator
  fullnameOverride: jaeger-operator
  image:
    repository: <OUR DOCKER REGISTRY>/jaegertracing/jaeger-operator
    tag: 1.21.3
    pullPolicy: IfNotPresent
    imagePullSecrets: []`

**Reply by jpkrohling** [reply]
If you want to customize the image definition, you can pass the flags to the operator: 

```console
$ podman run jaegertracing/jaeger-operator start --help | grep image
      --jaeger-agent-image string                       The Docker image for the Jaeger Agent (default "jaegertracing/jaeger-agent")
      --jaeger-all-in-one-image string                  The Docker image for the Jaeger all-in-one (default "jaegertracing/all-in-one")
      --jaeger-cassandra-schema-image string            The Docker image for the Jaeger Cassandra Schema (default "jaegertracing/jaeger-cassandra-schema")
      --jaeger-collector-image string                   The Docker image for the Jaeger Collector (default "jaegertracing/jaeger-collector")
      --jaeger-es-index-cleaner-image string            The Docker image for the Jaeger Elasticsearch Index Cleaner (default "jaegertracing/jaeger-es-index-cleaner")
      --jaeger-es-rollover-image string                 The Docker image for the Jaeger Elasticsearch Rollover (default "jaegertracing/jaeger-es-rollover")
      --jaeger-ingester-image string                    The Docker image for the Jaeger Ingester (default "jaegertracing/jaeger-ingester")
      --jaeger-query-image string                       The Docker image for the Jaeger Query (default "jaegertracing/jaeger-query")
      --jaeger-spark-dependencies-image string          The Docker image for the Spark Dependencies Job (default "jaegertracing/spark-dependencies")
      --openshift-oauth-proxy-image string              The Docker image location definition for the OpenShift OAuth Proxy (default "openshift/oauth-proxy:latest")
      --openshift-oauth-proxy-imagestream-name string   The name for the OpenShift OAuth Proxy imagestream
      --openshift-oauth-proxy-imagestream-ns string     The namespace for the OpenShift OAuth Proxy imagestream
```

**Reply by andreacoccodi** [reply]
does this translate in env vars created for the jaeger-operator deployment? if that's the case, the only missing thing would be a var in the values.yaml to define extra env vars for the deployment.

**Reply by jpkrohling** [reply]
I believe so: https://github.com/jaegertracing/jaeger-operator/blob/b98361cf4a8fc05e558001b310114aa5e57f8c35/cmd/root.go#L62

You can try it out locally, by setting `JAEGER_COLLECTOR_IMAGE` and creating a CR with the production strategy. It should use the image you specified in the env var.

**Reply by andreacoccodi** [reply]
sounds good, thanks
